gordo/machine/dataset/filter_periods.py

```diff
@@ -0,0 +1,205 @@
+from pprint import pformat
+import pandas as pd
+from sklearn.ensemble import IsolationForest
+from sklearn.preprocessing import MinMaxScaler
+from gordo.machine.dataset.datasets import pandas_filter_rows
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class WrongFilterMethodType(TypeError):
+    pass
+
+
+class filter_periods:
```

Comment:
According to [PEP8](https://www.python.org/dev/peps/pep-0008/#class-names) spec class names (except for build-in classes) should have CapWords names
```suggestion
class FilterPeriods:
```